### ISSUE TITLE
Core choice reverts to Gambatte after making a movie

### ISSUE THREAD

**Issue by ThunderAxe31** [TASVideos/BizHawk#2246]
Just as title. Happens in latest devbuild, https://github.com/TASVideos/BizHawk/commit/6371766130ff7cdccd55d8d793cc14459ce370f4

Here is a more detailed repro example:
1. Go on Config > Cores > GB > and set the core to either GBHawk or SubGBHawk
2. Reboot core
3. Record a movie
4. Close the movie
5. If you go on Config > Cores > GB, you'll see that the preference is back to Gambatte. In fact, if you reboot core, BizHawk will boot Gambatte.

**Comment by adelikat** [comment]
Is the movie correctly set to GBHawk in its sync settings?

**Comment by ThunderAxe31** [comment]
Yes, everything reflects the core name displayed in the bottom of the BizHawk window. It's just that sometimes it stealthly sets back to Gambatte by itself, and it's annoying.

By the way, I tried testing it again and now it's not reverting back to Gambatte anymore when I reboot the core, but the setting gets still changed by itself for no reason.

**Comment by alyosha-tas** [comment]
I tested this in every way I could think of with combinations of core setting changes, reboots, and movies recording / stopping / loading and couldn't get anything like what is described to happen. Do you still get this problem in the dev build @ThunderAxe31  ? Any more detailed repro steps I could try?

**Comment by alyosha-tas** [comment]
I tried again with a fresh dev build, and there are rare occasions that I get the error as described. I'm working on exact repro steps.

**Comment by alyosha-tas** [comment]
Alright here is what I could come up with.

If you: 

1. have gambatte selected in the core selection options and load a rom (running core is Gambatte)
2. open a GBHawk movie (running core is now GBHawk)
3. stop that movie (still in GBHawk)
4. record a new movie (even though you are recording a new movie, the core does not change back to Gambatte)

This is the only situation I can get something similar to what the issue describes to happen.

**Comment by adelikat** [comment]
It should be fixed now by a recent change by natt, try again

**Comment by alyosha-tas** [comment]
Not fixed. Same repro. steps produce the same result.

**Comment by nattthebear** [comment]
I'm able to reproduce with alyosha's steps.  This seems to be some sort of movie settings confusion.  When we get to step 4, `MainForm.CoreSyncSettings` provides RomLoader a set of GBHawk syncsettings, that appear to come from the old movie.  This causes Gambatte to crash on load and the RomLoader to fall back to GBHawk, which does successfully load with those same syncsettings.

**Comment by adelikat** [comment]
TAStudio isn't free of this problem either:

1) Set core preference to gambatte, and reboot core
2) Set core preference to GBHawk, and do NOT reboot core
3) Open TAStudio

You will have a default.tasproj opened using Gambatte as the core.  Clicking New project, you will still have Gambatte

**Comment by YoshiRulz** [comment]
Reproduced with [alyosha's steps](https://github.com/TASVideos/BizHawk/issues/2246#issuecomment-667676147) on `master` at 995993357. With the same steps on 2.4.2, the sync settings mismatch [described by natt](https://github.com/TASVideos/BizHawk/issues/2246#issuecomment-674534678) is shown in a dialog and the rom is closed.

**Comment by YoshiRulz** [comment]
Repro'd on 2.8. New behaviour is that when recording a new movie with GBHawk loaded, the preference for Gambatte is ignored, and the rom reloads in GBHawk. At no point is Gambatte instantiated and there is no error in the console or GUI. The user's preference is not changed; Gambatte will be used if the movie is stopped and the rom is reloaded. I'm willing to call the remaining weirdness wontfix.
